### PR TITLE
JBPM-5529 Dead end after adding subform/multiple subform into the canvas.

### PR DESCRIPTION
**Pull request by alessioscalici**
**WARNING**: twin PR https://github.com/kiegroup/kie-wb-common/pull/1042

@pefernan @jsoltes could you review?

### CODE REVIEW

File: kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/fieldTypes/relations/multipleSubform/definition/MultipleSubFormFieldDefinition.java [hunk 1]
```diff
@@ -53,7 +53,6 @@
     @SelectorDataProvider(
             type = SelectorDataProvider.ProviderType.REMOTE,
             className = "org.kie.workbench.common.forms.editor.backend.dataProviders.VFSSelectorFormProvider")
-    @NotEmpty
```
Comment (pefernan) [inline kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/fieldTypes/relations/multipleSubform/definition/MultipleSubFormFieldDefinition.java]: Unused import

File: kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/fieldTypes/relations/subForm/definition/SubFormFieldDefinition.java [hunk 2]
```diff
@@ -51,8 +51,6 @@
     @SelectorDataProvider(
             type = SelectorDataProvider.ProviderType.REMOTE,
             className = "org.kie.workbench.common.forms.editor.backend.dataProviders.VFSSelectorFormProvider")
-    @NotNull
-    @NotEmpty
```
Comment (pefernan) [inline kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/fieldTypes/relations/subForm/definition/SubFormFieldDefinition.java]: Unused import

File: kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/fieldTypes/relations/subForm/definition/SubFormFieldDefinition.java [hunk 3]
```diff
@@ -51,8 +51,6 @@
     @SelectorDataProvider(
             type = SelectorDataProvider.ProviderType.REMOTE,
             className = "org.kie.workbench.common.forms.editor.backend.dataProviders.VFSSelectorFormProvider")
-    @NotNull
```
Comment (pefernan) [inline kie-wb-common-forms/kie-wb-common-forms-core/kie-wb-common-forms-fields/src/main/java/org/kie/workbench/common/forms/fields/shared/fieldTypes/relations/subForm/definition/SubFormFieldDefinition.java]: Unused import

**Comment by manstis** [comment]
jenkins please execute full downstream build

**Comment by manstis** [comment]
jenkins please execute full downstream build

**Pull request merged**